plugins/builddep.py

```diff
@@ -135,13 +135,13 @@ def _rpm_dep2reldep_str(rpm_dep):
         return rpm_dep.DNEVR()[2:]
 
     def _install(self, reldep_str):
-        try:
-            self.base.install(reldep_str)
-        except dnf.exceptions.MarkingError:
-            msg = _("No matching package to install: '%s'")
-            logger.warning(msg, reldep_str)
+        sltr = dnf.selector.Selector(self.base.sack)
+        sltr = sltr.set(provides=reldep_str)
+        if sltr.matches():
+            self.base._goal.install(select=sltr)
```

Comment:
I'm not sure `strict` would necessarily be defined to act in the manner you suggest...